### ISSUE TITLE
HCL2 - vault_aws_engine

### ISSUE THREAD

**Issue by dtroup-global** [hashicorp/packer#11536]
TLDR: allow using the `use_vault_aws_engine` in an aws datasource.

Hi, having an issue with the HCL2 formatted files. I’m converting a working json file to hcl, and using vault as part of the build. When converting the file it puts in an entry for the AWS authentication but it does not know how to handle the vault_aws_auth entry.

#### Reproduction Steps

packer validate base.json
packer build base.json

Both of these work

packer hcl2_upgrade base.json
fix a couple of issues with commands (change them to locals)
packer validate base.pkr.hcl

### Packer version

Packer v1.7.9

### Simplified Packer Template
```
data "amazon-ami" "ubuntu" {
  filters = {
    name                = "ubuntu-minimal/images/hvm-ssd/ubuntu-bionic-18.04-amd64-minimal-*"
    root-device-type    = "ebs"
    virtualization-type = "hvm"
  }
  most_recent = true
  owners      = ["aws-marketplace"]
  region      = "eu-west-1"
  vault_aws_engine = {
    name = "myrole"
  }
}

packer {
  required_plugins {
    amazon = {
      version = "= 1.0.7"
      source  = "github.com/hashicorp/amazon"
    }
    inspec = {
      version = "= 1.0.0"
      source  = "github.com/hashicorp/inspec"
    }
    ansible = {
      version = "= 1.0.1"
      source  = "github.com/hashicorp/ansible"
    }
  }
}
```
### Operating system and Environment details

Tested on ubuntu 18.04 and Mac OS

### Log Fragments and crash.log files
```
Error: Unsupported argument

  on base.pkr.hcl line 44:
  (source code not available)

An argument named "vault_aws_engine" is not expected here. Did you mean to
define a block of type "vault_aws_engine"?
```
I also logged a discussion on the forum, here (https://discuss.hashicorp.com/t/hcl2-vault-aws-engine/34841) 
but so far no replies :(

**Comment by azr** [comment]
Hello there, thanks for opening. The `hcl2_upgrade` tool can sometimes not know what to do with types that differ to much.
But the fix should be quick for you :

> Did you mean to define a block of type "vault_aws_engine"?

Did that work ? Something like:

```hcl
vault_aws_engine {
 // stuff
}
```

**Comment by dtroup-global** [comment]
Hi @azr  yes, I've made that change and removed the = the validate does now pass, however, I now get a different error when running the build.

 Error: Datasource.Execute failed: Error querying AMI: AuthFailure: AWS was not able to validate the provided access credentials
status code: 401, request id: 59bae08d-4492-4b03-9427-cb97bd14e256

on base.pkr.hcl line 35:
(source code not available)

According to the documentation for ebs:

access_key, secret_key and region state:

On EBS, this is not required if you are using use_vault_aws_engine for authentication instead.

But I still can't see where to set use_vault_aws_engine in any documentaiton

**Comment by azr** [comment]
Eh, weird, I also can't find the documentation for the `use_vault_aws_engine` field in the docs ! But it should be settable in the top ebs builder block. This option won't be available in the datasource yet though, so you will have to pass the auth info manually. A better option if you have the time would be to make that option available in the datasource.

Closing that issue for now, as it seems to be solved, let us know if something's up !
